internal/applicationsnapshot/input.go

```diff
@@ -17,61 +17,140 @@
 package applicationsnapshot
 
 import (
+	"context"
 	"errors"
 	"fmt"
 
 	"github.com/ghodss/yaml"
 	app "github.com/redhat-appstudio/application-api/api/v1alpha1"
 	log "github.com/sirupsen/logrus"
 	"github.com/spf13/afero"
+	"golang.org/x/exp/slices"
+
+	"github.com/hacbs-contract/ec-cli/internal/kubernetes"
+	"github.com/hacbs-contract/ec-cli/internal/utils"
 )
 
-func DetermineInputSpec(fs afero.Fs, filePath string, input string, imageRef string) (*app.SnapshotSpec, error) {
-	var appSnapshot app.SnapshotSpec
+const unnamed = "Unnamed"
+
+type Input struct {
+	File     string
+	JSON     string
+	Image    string
+	Snapshot string
+}
+
+type snapshots struct {
+	app.SnapshotSpec
+}
+
+func (s *snapshots) merge(snap app.SnapshotSpec) {
+	if s.Application == "" {
+		s.Application = snap.Application
+	}
 
-	// read ApplicationSnapshot provided as a file
-	if len(filePath) > 0 {
-		content, err := afero.ReadFile(fs, filePath)
+	if s.DisplayName == "" {
+		s.DisplayName = snap.DisplayName
+	}
+
+	if s.DisplayDescription == "" {
+		s.DisplayDescription = snap.DisplayDescription
+	}
+
+	images := map[string]string{}
+	for _, c := range s.Components {
+		images[c.ContainerImage] = c.Name
+	}
+	for _, c := range snap.Components {
+		if name, ok := images[c.ContainerImage]; !ok || name == "" || name == unnamed {
+			if ok {
+				images[c.ContainerImage] = c.Name
+				i := slices.IndexFunc(s.Components, func(x app.SnapshotComponent) bool {
+					return x.ContainerImage == c.ContainerImage
+				})
+				s.Components[i].Name = c.Name
+			} else {
+				images[c.ContainerImage] = c.Name
+				s.Components = append(s.Components, c)
+			}
+		}
+	}
+}
+
+func DetermineInputSpec(ctx context.Context, input Input) (*app.SnapshotSpec, error) {
```

Comment:
IMO, the merging functionality is confusing from a user's perspective. The user must proactively know about it, and understand the precedence logic. I'd be in favor of just throwing an error if the user tries to use more than one source.

This comment is highly subjective, so feel free to ignore it :)

But if we do keep the merge logic, let's make sure it's clear how it behaves from the ec-cli help.

Reply:
@hacbs-contract/devs thoughts?

Reply:
Let's keep it if no one else has an opinion.